### PR TITLE
seq2seq speech recognition example: use forward attention mask if apply spec augment is True

### PR DESCRIPTION
**Pull request by sorgfresser**
# What does this PR do?

The example for speech recognition for `seq2seq` did not forward the attention mask for using SpecAugment on using whisper. I strongly believe this is a bug in the example and it should be resolved now.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@patrickvonplaten 
@bofenghuang

### CODE REVIEW

**Comment by amyeroberts** [comment]
cc @sanchit-gandhi

File: examples/pytorch/speech-recognition/run_speech_recognition_seq2seq.py [hunk 1]
```diff
@@ -460,7 +460,7 @@ def main():
     # if SpecAugment is used for whisper models, return attention_mask to guide the mask along time axis
     forward_attention_mask = (
         getattr(config, "model_type", None) == "whisper"
-        and getattr(config, "apply_spec_augment", False)
+        and getattr(config, "apply_spec_augment", True)
```
Comment (sanchit-gandhi) [inline examples/pytorch/speech-recognition/run_speech_recognition_seq2seq.py:463]: Hey @sorgfresser - this is just checking whether the model config has the attribute `apply_spec_augment`:
* If the config has the attribute `apply_spec_augment`, then we return the value in the config
* If it does **not** have the attribute `apply_spec_augment`, then we return `False`

The original Whisper models were **not** trained with Spec Aug, thus if they do not have the attribute `apply_spec_aug` in the config, the code defaults to not using Spec Aug (`False`). So the code is correct in its current state

If you want to train the model with Spec Aug enabled, pass the argument `--apply_spec_augment` when you launch training, and the config attribute `apply_spec_aug` will be set to `True`.

=> this will enable Spec Aug in the fine-tuning run
